src/components/provider/_fields/proxy-select/proxy-select-field.js

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Field } from 'redux-form';
+import { intlShape, injectIntl } from 'react-intl';
+
+import { Select } from '@folio/stripes-components';
+import styles from './proxy-select-field.css';
+
+function ProxySelectField({ proxyTypes, rootProxy, intl }) {
+  let proxyTypesRecords = proxyTypes.resolver.state.proxyTypes.records;
+  let rootProxyId = rootProxy.data.attributes.proxyTypeId;
+  let options = [];
+
+  if (proxyTypesRecords && rootProxyId) {
+    for (let proxyTypesRecord in proxyTypesRecords) {
+      if (Object.prototype.hasOwnProperty.call(proxyTypesRecords, proxyTypesRecord)) {
+        let selectValue = proxyTypesRecords[proxyTypesRecord].attributes.id;
+        if (rootProxyId === selectValue) {
+          options.push({ label: `${intl.formatMessage({ id: 'ui-eholdings.provider.inherited' })}-${proxyTypesRecords[proxyTypesRecord].attributes.name}`,
+            value: proxyTypesRecords[proxyTypesRecord].attributes.id });
+        } else {
+          options.push({ label: proxyTypesRecords[proxyTypesRecord].attributes.name,
+            value: proxyTypesRecords[proxyTypesRecord].attributes.id });
+        }
+      }
+    }
+  }
+
+  return (
+    <div
+      data-test-eholdings-provider-proxy-select-field
+      className={styles['proxy-select-field']}
+    >
+      <Field
+        id="eholdings-providers-proxy-id"
+        name="proxyId"
+        component={Select}
+        dataOptions={options}
+        label={intl.formatMessage({ id: 'ui-eholdings.provider.proxy' })}
+        disabled={options.length < 2}
```

Comment:
If we disable the dropdown, would the user be necessarily aware as to why it is disabled? Not sure if this is an obvious user experience? We should get feedback from others regarding this.

Reply:
I'll follow up with Khalilah -- we did discuss this and the case where all packages for the provider are not selected. In both of these cases the user should not be able to change the proxy selection

Reply:
Maybe there could be some messaging around the disabled drop-down. Saying something like "No proxies available"